src/network/networkd-ndisc.c

```diff
@@ -75,6 +77,30 @@ static void ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
                 return;
         }
 
+        SET_FOREACH(address, link->addresses, i) {
+                if (!memcmp(&gateway, &address->in_addr,
+                            sizeof(address->in_addr))) {
```

Comment:
Yes, this must be fixed. I'm apparently a bit rusty since I failed to pay attention to this. v2 coming up.